Content.Server/Disease/Effects/DiseaseProgression.cs

```diff
@@ -0,0 +1,52 @@
+using Content.Server.Disease.Components;
+using JetBrains.Annotations;
+using Content.Shared.Disease;
+using Robust.Shared.IoC;
+
+namespace Content.Server.Disease.Effects
+{
+    /// <summary>
+    /// Handles a disease which incubates over a period of time
+    /// before adding another component to the infected entity
+    /// currently used for zombie virus
+    /// </summary>
+    [UsedImplicitly]
+    public sealed class DiseaseProgression : DiseaseEffect
+    {
+        /// <summary>
+        /// The rate that's increased over time. Defaults to 1% so the probability can be varied in yaml
+        /// </summary>
+        [DataField("rate")]
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float Rate = 0.01f;
+
+        /// <summary>
+        /// The current amount of progression that has built up.
+        /// </summary>
+        [DataField("progression")]
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float Progression = 0.00f;
+
+		/// <summary>
+		/// The component that is added at the end of build up
+		/// </summary>
+		[DataField("comp")]
+		public string? Comp =  null;
+
+        public override void Effect(DiseaseEffectArgs args)
+        {
+            if ( Progression < 1) //increases steadily until 100%
+            {
+                Progression += Rate;
+            }
```

Comment:
uhhh im almost positive this doesnt work or will work weirdly, don't store data that should last across ticks on a disease effect like this (or similar things). ensure a component on the entity and modify the progression in an update loop.

Reply:
actgually just update it in this loop the same way but have the progression stored on the comp instead